modules/internal/ChapelBase.chpl

```diff
@@ -1215,6 +1215,9 @@ module ChapelBase {
     delete _to_unmanaged(e);
   }
 
+  pragma "no doc"
```

Comment:
This module isn't documented, so "no doc" isn't needed. Can we put `chpl_comm_task_create` with `chpl_comm_task_end`?

Reply:
Can we put a doc on on both task_create and task_end saying something like:


create -- "called by the parent task before child tasks are spawned" 
end -- "called by the child task while ending"

at first it wasn't obvious to me that task_create was supposed to be called by the parent or the child I wonder if chpl_comm_task_pre_create is slightly more obvious in that regard, but i don't think it's worth changing

Reply:
Ah, I see these are documented in the runtime already, maybe that's enough

Reply:
Ah, I think I actually stole a decl from `MemConsistency.chpl` to form that one, is why the "no doc" is there.  I'll remove it.

The decl and use of `chpl_comm_task_end()` are below this point in ChapelBase, so we'd actually want to move `task_end` to be with `task_create` in order to put them together.  But yeah, that would probably be good.

Reply:
I don't think chapel requires functions to be declared before they're used, but up to you where you want to place the externs.